### PR TITLE
Fixed instrumentation code so that it can deal with:

### PR DESCRIPTION
**Pull request by odersky**
1. leading imports: object sheet { import math; … }
2. trailing objects: object sheet { … } object other { … }

Review by @dragos

### CODE REVIEW

**Comment by odersky** [comment]
The failures look almost certainly unrelated to the patch. @xeno.by @dragos can you troubleshoot please? This fix is crucial for the Coursera class and I have no time to follow up on this myself.

**Comment by xeno-by** [comment]
Yes I'm already on it

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by gkossakowski** [comment]
Hmmm. I thought worksheet stuff should be in 2.9.x branch that coursera is using, no?

**Comment by odersky** [comment]
Should be in both 2.9 and 2.10. Will have to be backported to 2.9.

**Comment by dragos** [comment]
LGTM

File: src/compiler/scala/tools/nsc/interactive/ScratchPadMaker.scala [hunk 1]
```diff
@@ -35,7 +35,7 @@ trait ScratchPadMaker { self: Global =>
 
     private def literal(str: String) = "\"\"\""+str+"\"\"\""
 
-    private val prologue = "import scala.runtime.WorksheetSupport._; def main(args: Array[String])=$execute{"
+    private val prologue = ";import scala.runtime.WorksheetSupport._; def main(args: Array[String])=$execute{"
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/interactive/ScratchPadMaker.scala:38]: Can there be arbitrary user code in scope here? Then it should be:

``` scala
;import _root_.scala.runtime.WorksheetSupport._; def main(args: _root_.scala.Array[_root_.scala.String])
```

Comment (paulp) [inline src/compiler/scala/tools/nsc/interactive/ScratchPadMaker.scala:38]: Ha ha, lot of good that will do you.

https://issues.scala-lang.org/browse/SI-6217

```
package foo
package _root_

object Test {
  import _root_.scala.Option
}
```

Comment (retronym) [inline src/compiler/scala/tools/nsc/interactive/ScratchPadMaker.scala:38]: Heh, reminds me of this pearl:

``` scala
scala> import collection.{mutable => _, _}; `_`.Buffer(0)
import collection.{mutable=>_, _}
res0: scala.collection.mutable.Buffer[Int] = ArrayBuffer(0)
```

(Just lodged as: https://issues.scala-lang.org/browse/SI-6426)

**Comment by gkossakowski** [comment]
PLS REBUILD ALL

**Pull request merged**